src/scripts/add_models_to_leaderboard.py

```diff
@@ -0,0 +1,207 @@
+        search: Dict[str, List[Dict[str, Any]]] = {task_name: []}
+        for language in languages:
+            # Get supertask, and check that it is correct.
+            if task_config.supertask == "sequence-classification":
+                supertask = "text-classification"
+            else:
+                supertask = task_config.supertask
+
+            # Add search terms, some tasks have no search terms, as they are examples of general supertasks.
+            if not task_config.search_terms:
+                search[task_name].append(
+                    {
+                        "search": "",
+                        "filter": ModelFilter(task=supertask, language=language),
+                    }
+                )
+            else:
+                for search_term in task_config.search_terms:
+                    search[task_name].append(
+                        {
+                            "search": search_term,
+                            "filter": ModelFilter(task=supertask, language=language),
+                        }
+                    )
+
+        searches.append(search)
+    return searches
+
+
+def prepare_cache_and_get_succeeded_and_failed_models(
+    cache_dir: str, output_path: str
+) -> Tuple[Any, bool, List[str], Any]:
+    """
+    Prepare cache and get succeeded and failed models.
+
+    Args:
+        cache_dir (str): Path to cache directory.
+        output_path (str): Path to output directory.
+
+        Returns:
+        failed_models_csv_writer (csv.writer): Writer for failed_models.csv.
+        failed_models_csv_is_new (bool): True if failed_models.csv is new, False if it already existed.
+        models_ids_evaluated (list): List of model ids which have already been evaluated.
+        evaluated_models_csv_writer (csv.writer): Writer for evaluated_models.csv.
+    """
+    # check if output_path exists, if not, create it.
+    if not os.path.exists(output_path):
+        os.makedirs(output_path)
+
+    # check if cache_dir exists, if not, create it.
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+
+    # make folder in cache_dir for evaluated models list
+    if not os.path.exists(f"{cache_dir}/evaluated_models"):
+        os.makedirs(f"{cache_dir}/evaluated_models")
+
+    # Check if we already have a list of failed models, if so, we load it.
+    failed_models_csv_is_new = False
+    try:
+        failed_models_csv = open(f"{output_path}/failed_models.csv", "a")
+    except FileNotFoundError:
+        failed_models_csv = open(f"{output_path}/failed_models.csv", "w+")
+
+    # Check if we already have a list of evaluated_models, if so, we load it.
+    try:
+        models_ids_evaluated_csv = open(
+            f"{cache_dir}/evaluated_models/evaluated_models.csv", "a"
+        )
+    except FileNotFoundError:
+        models_ids_evaluated_csv = open(
+            f"{cache_dir}/evaluated_models/evaluated_models.csv", "w+"
+        )
```

Comment:
This is assuming Linux. To make paths compatible I'd suggest we use `pathlib`. Then a path can be defined via, e.g., `Path("location") / "of" / "the" / "file"`. Instead of `open(...)` you then use `path.open(...)`.